### ISSUE TITLE
Mkdocs are failing in the CI

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-spec#661]
We're getting the following error when the CI runs (see PR #659 for an example):

```
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1[6](https://github.com/spdx/spdx-spec/runs/6116293939?check_suite_focus=true#step:5:6)5[7](https://github.com/spdx/spdx-spec/runs/6116293939?check_suite_focus=true#step:5:7), in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/mkdocs/__main__.py", line 1[8](https://github.com/spdx/spdx-spec/runs/6116293939?check_suite_focus=true#step:5:8)7, in build_command
    build.build(config.load_config(**kwargs), dirty=not clean)
  File "/usr/local/lib/python3.[10](https://github.com/spdx/spdx-spec/runs/6116293939?check_suite_focus=true#step:5:10)/site-packages/mkdocs/config/base.py", line 216, in load_config
    from mkdocs.config.defaults import get_schema
  File "/usr/local/lib/python3.10/site-packages/mkdocs/config/defaults.py", line 1, in <module>
    from mkdocs.config import config_options
  File "/usr/local/lib/python3.10/site-packages/mkdocs/config/config_options.py", line 8, in <module>
    from mkdocs import utils, theme, plugins
  File "/usr/local/lib/python3.10/site-packages/mkdocs/theme.py", line 6, in <module>
    from mkdocs.utils import filters
  File "/usr/local/lib/python3.10/site-packages/mkdocs/utils/filters.py", line [13](https://github.com/spdx/spdx-spec/runs/6116293939?check_suite_focus=true#step:5:13), in <module>
    @jinja2.contextfilter
AttributeError: module 'jinja2' has no attribute 'contextfilter'
Error: Process completed with exit code 1.
```

**Comment by NorioKobota** [comment]
Hello,
I think it is inconvenient not to be able to see the spec website, so I'd like to send a PR  to all necessary branches.
If okay, could you tell me the following information?
- The branch name to be referenced in spdx.github.io/spdx-spec/. => development/v 2.2.2 or master?
- The branch name to refer to in spdx.github.io/spdx-spec/v2-draft/ => development/v 2.3?
- The branch name to refer to in spdx.github.io/spdx-spec/v3-draft/ => development/v 3.0?
Best regards.

**Comment by goneall** [comment]
@tsteenbe @kestewart  - Can you answer the above questions?  I'm not sure where the merges are for the 2.3 development, so I'm not sure if the current is master or development/v2.2.2.

Also, what do we want to do about development/v3.0?  It is not very current and may be confusing if we leave this up.  I I had more time, I'd add some documentation to the development/v3.0 branch and point them to https://github.com/spdx/spdx-3-model

@NorioKobota Thank you very much for helping with this.

> The branch name to refer to in spdx.github.io/spdx-spec/v2-draft/ => development/v2.3

This should be development/v2.3

**Comment by goneall** [comment]
Accidentally closed during tech call - re-opened

@NorioKobota Thanks again for helping with this.

**Comment by goneall** [comment]
@NorioKobota Just to summarize answers to your above questions based on the tech call:

> The branch name to be referenced in spdx.github.io/spdx-spec/ => development/v2.2.2 or master?

master

> The branch name to refer to in spdx.github.io/spdx-spec/v2-draft/ => development/v2.3?

development/v2.3

> The branch name to refer to in spdx.github.io/spdx-spec/v3-draft/ => development/v3.0?

This should be removed - we're dropping the development/v3.0 branch.

**Comment by NorioKobota** [comment]
@goneall @kestewart 
Thank you. I got it.
Since the current master branch is synchronized with the v2.2.2 branch, I'll send PR to both the development/v2.2.2 branch and the master branch.
The v3 branch build has already been removed with following commit.
https://github.com/spdx/spdx-spec/commit/693f14bfed7acf5cf1e3bdbc420bd4f0c4a93b88

**Comment by goneall** [comment]
@NorioKobota I merged in the PR's - however, I'm still not seeing the updated site.

It may be related to the 3.0 branch not executing.

Could you make one more PR to the 3.0 branch?  I'll review and merge once it is created and we'll se if the docs get regenerated.

**Comment by NorioKobota** [comment]
@goneall Yes, I checked it. The reason is that the build setting of the v3.0 branch remains in the publish.yml of the master branch. I will send you a PR later.
Thanks.

**Comment by goneall** [comment]
@NorioKobota Thanks for the PR's.  I merged them and the CI is now running without error :)

The main spec page spdx.github.io/spdx-spec/ is correct.

However, the v2 drafts page is still pointing to v2.2.2 when it should be v3.

I'll take a look and see if I can figure out why.

**Comment by NorioKobota** [comment]
@goneall Thank you.
I certainly forgot to correct it.
https://github.com/spdx/spdx-spec/commit/6fb512276970e1d11c465ae972c3dc4529780a40
It works correctly with this correction.

**Comment by NorioKobota** [comment]
BTW, In fact, this https://github.com/spdx/spdx-spec/commit/6fb512276970e1d11c465ae972c3dc4529780a40 modification only affects when you commit/push the master/v2.2.2 branch. (Therefore, when you commit/push to the v2.3 branch, Github Pages will be successfully deployed as it is.)
I think it's better to limit the watch branch to only the active branch, such as the development/v2.3 branch, in the 'on' section of publish.yml. 

> https://github.com/spdx/spdx-spec/blob/development/v2.3/.github/workflows/publish.yml#L3
>  "branches: [master, development/v2.3]" should be "branches: [development/v2.3]"

Otherwise, if something goes wrong, as in this case, we'll be maintaining the same file separately in different branches.
If you have time, I would appreciate it if you could discuss it on tech meeting.
And if you need it, I will post this to tech ML, so please let me know.

Best,
--kobota

**Comment by goneall** [comment]
@tsteenbe Since you were the original author of the Mkdocs - can you comment on @NorioKobota observation above?

Seems to make sense to limit the branch being updated.  The only issue I can think of is we need to update this whenever we create a new branch.  Probably something we can add into a release checklist.

@iamwillbar Also - any thoughts on this?

**Comment by goneall** [comment]
This is resolved with PR #702 PR #701 PR #699 PR #698 PR #697 PR #693 PR #692 and/or PR #676 

Thanks @NorioKobota and @tsteenbe with the help - let me know if you see any remaining issues
